docs/blueprints/referral-manager-v1.md

```diff
@@ -0,0 +1,229 @@
+# Referral Manager V1: Cross-Satellite Referral Link
+
+## Abstract
+
+This document describes how to handle user referrals across satellites.
+
+## Background
+
+Previous referral program only works per satellite. This means a user on us-east-1 satellite can only refer people to join us-east-1.
+
+We want referrals to be valid on all Tardigrade satellites. This will allow the user to select the satellite that is best for them.
+
+### Non-Goals
+A registration page where a user can select their preferred Satellite during registration.
+
+## Design
+
+### Definitions
+
+Referral Manager
+: A stand-alone process running outside of satellites. It should only talk to tardigrade branded satellites.
+
+Referral Manager CLI
+: A CLI that will allow operators to invoke referral link generation.
+
+Referral Link
+: A one-time-use URL that contains a unique invitation token.
+
+Invitation Token
+: A random number generated by Referral Manager for each user in the referral program.
+
+Host Satellite
+: The satellite that a particular token is stored on before it is redeemed.
```

Comment:
What does it mean for a token to be redeemed? Does that mean someone is signing up for an account using a `referral link` that somebody else gave them? Can tokens be redeemed only once, then? I'm having some trouble picturing the structure of the overall referral program. If I want Fred Rogers to sign up, do I need to go to the satellite web interface to request that a new fresh `invitation token` be generated, so I can give it to Fred? I don't just get a "referral URL" that I can give to everyone I want?

Reply:
discussed in our call